src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -526,6 +576,7 @@ public BootstrapTaskStatus next() {
             throw new NoSuchElementException();
         }
         cursor++;
-        return bootstrapTaskStatusList.get(cursor - 1);
+        this.activeTask = bootstrapTaskStatusList.get(cursor - 1);
```

Comment:
Modifying the active task in here seems sketchy. Why do you need to do this? Caller of next can set it if desired.

Reply:
I've implemented a method `getUnstartedTasks()` which helps determine which components need to execute bootstrap-on-rollback.  `getUnstartedTasks()` needs to exclude the bootstrap task which just failed (which I call the "active task"), so I created a member named `activeTask` and set it whenever we fetch a task from the iterator.